### PR TITLE
10.0 Port account_fiscal_position_vat_check

### PR DESCRIPTION
**Pull request by alexis-via**

### CODE REVIEW

File: account_fiscal_position_vat_check/__manifest__.py [hunk 1]
```diff
@@ -1,62 +1,18 @@
...
-##############################################################################
-
+# -*- coding: utf-8 -*-
+# © 2013-2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 {
     'name': 'Account Fiscal Position VAT Check',
-    'version': '8.0.0.1.1',
+    'version': '10.0.0.1.1',
```
Comment (lmignon) [inline account_fiscal_position_vat_check/__manifest__.py]: 10.0.1.0.0

**Review by lmignon — CHANGES_REQUESTED** [review]
A little change to start with a proper version number :smirk:

File: account_fiscal_position_vat_check/models/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+from odoo import models, api, _
+from odoo.exceptions import UserError
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def action_move_create(self):
+        '''Check that the customer has VAT set
```
Comment (pedrobaeza) [inline account_fiscal_position_vat_check/models/account_invoice.py]: Docstrings uses triple double-quote

File: account_fiscal_position_vat_check/models/partner.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+from odoo import models, api, _
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.onchange('property_account_position_id')
+    def fiscal_position_change(self):
+        '''Warning if the fiscal position requires a VAT number and the
```
Comment (pedrobaeza) [inline account_fiscal_position_vat_check/models/partner.py]: Double quote

File: account_fiscal_position_vat_check/models/partner.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        partner doesn't have one yet'''
+        fp = self.property_account_position_id
+        if fp and self.customer and not self.vat:
```
Comment (pedrobaeza) [inline account_fiscal_position_vat_check/models/partner.py]: You can save one `if` putting directly `if fp.vat_required` in the first condition

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: account_fiscal_position_vat_check/models/partner.py [hunk 5]
```diff
@@ -10,17 +10,16 @@ class ResPartner(models.Model):
...
-            if fp.vat_required:
-                return {
-                    'warning': {
-                        'title': _('Missing VAT number:'),
-                        'message': _(
-                            "You have set the fiscal position '%s' "
-                            "that require the customer to have a VAT number, "
-                            "but the VAT number is missing.") % fp.name
-                    }
+        if fp and fp.vat_required and self.customer and not self.vat:
```
Comment (pedrobaeza) [inline account_fiscal_position_vat_check/models/partner.py]: You don't need to check `fp`, as `fp.vat_required` will be False if fp is an empty recordset, but not blocking.

**Review by pedrobaeza — APPROVED** [review]

**Review by lmignon — APPROVED** [review]
Code review only

**Pull request merged**